charts/seed-controlplane/charts/kube-apiserver/templates/network-policy.yaml

```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: kube-apiserver-deny-blacklist
+  namespace:  {{ .Release.Namespace }}
+spec:
+  podSelector:
+    matchLabels:
+      app: kubernetes
+      role: apiserver
+  policyTypes:
+  - Egress
+  egress:
+  - to:
+    - ipBlock:
+        # Allow all except seed networks
+        cidr: 0.0.0.0/0
+        except:
+        - {{ .Values.seedNetworks.pod }}
```

Comment:
I see below:
Values.seedNetworks.pod == 10.0.0.0/24

So this range is not reachable from the apiserver.

But also:
shootNetworks.service: 10.0.0.0/24

This is the same IP range. So the apiserver cannot talk to the shoot services directly. Have you verified it never needs to do that? At the moment I can't think of any reason why it would need to.

Reply:
This are just default placeholder in values.yaml. Actual values come from seed.yaml and shoot.yaml while applying chart. So, please don't get confused with it. 

FYI, Gardener internally takes care that these networks don't conflict between seed and shoot (that too because of VPN setup sidecar). Otherwise apiserver need not talk to anything from shoot, neither pod nor services. kube-apiserver only needs access to etcd and kube-dns (which runs in seed network), for which we have other NetworkPolicy instance in same file.

Still thanks for pointing that out, may be I should simply keep those ranges separate to avoid such confusion.